plugins/by-name/copilot-lua/default.nix

```
@@ -1,172 +1,181 @@
+            `lib.getExe nodePackage` if `nodePackage` is not-null, otherwise `null`
+          '';
+          pluginDefault = "node";
+          example = lib.literalExpression "lib.getExe pkgs.nodejs";
           description = ''
-            Use this field to provide the path to a specific node version such as one installed by
-            `nvm`.
-            Node.js version must be 16.x or newer.
+            Define the node command to use for copilot-lua.
+            Node.js version must be 18.x or newer.
           '';
         };
 
-        serverOptsOverrides = helpers.defaultNullOpts.mkAttrsOf' {
-          type = types.anything;
-          pluginDefault = { };
-          description = ''
-            Override copilot lsp client `settings`.
-            The settings field is where you can set the values of the options defined in
-            https://github.com/zbirenbaum/copilot.lua/blob/master/SettingsOpts.md.
-            These options are specific to the copilot lsp and can be used to customize its behavior.
+      server_opts_overrides = defaultNullOpts.mkAttrsOf' {
+        type = types.anything;
+        pluginDefault = { };
+        description = ''
+          Override copilot lsp client `settings`.
+          The settings field is where you can set the values of the options defined in
+          https://github.com/zbirenbaum/copilot.lua/blob/master/SettingsOpts.md.
+          These options are specific to the copilot lsp and can be used to customize its behavior.
 
-            Ensure that the `name` field is not overridden as is is used for efficiency reasons in
-            numerous checks to verify copilot is actually running.
+          Ensure that the `name` field is not overridden as is is used for efficiency reasons in
+          numerous checks to verify copilot is actually running.
 
-            See `:h vim.lsp.start_client` for list of options.
-          '';
-          example = {
-            trace = "verbose";
-            settings = {
-              advanced = {
-                listCount = 10; # number of completions for panel
-                inlineSuggestCount = 3; # number of completions for getCompletions
-              };
+          See `:h vim.lsp.start_client` for list of options.
+        '';
+        example = {
+          trace = "verbose";
+          settings = {
+            advanced = {
+              listCount = 10; # number of completions for panel
+              inlineSuggestCount = 3; # number of completions for getCompletions
             };
           };
         };
       };
+    };
+
+  settingsExample = {
+    panel = {
+      enabled = true;
+      auto_refresh = true;
+    };
+    suggestion = {
+      enabled = true;
+      auto_trigger = false;
+      hide_during_completion = false;
+      debounce = 90;
+      keymap = {
+        accept_word = false;
+        accept_line = false;
+      };
+    };
+  };
+
+  extraOptions.nodePackage = lib.mkPackageOption pkgs "nodejs" {
```

To be clear, I'm fine with either.

But this is how we'd _normally_ format it, for extra clarity and to minimise diff size if additional options are added down the line

```suggestion
  extraOptions = {
    nodePackage = lib.mkPackageOption pkgs "nodejs" {
```